docs/build/build_system.md

```diff
@@ -0,0 +1,255 @@
+# Build system
+
+## Implementation details
+
+The project uses CMake as the build system generator. CMake allows for cross-platform builds and provides the necessary flexibility for both project development and its use in external projects.
```

Comment:
It was phrased this way since it is technically correct - we use Ninja / Make as our build system and CMake generates the definitions for us.
<img width="1281" alt="image" src="https://github.com/user-attachments/assets/62450892-f85b-4770-b254-80cc014d9aaf">

Reply:
I propose to insert link to CMake